cylc/flow/network/scan.py

```diff
@@ -239,17 +241,34 @@ def get_scan_items_from_fs(owner_pattern=None, reg_pattern=None):
 
             # Filter suites by name
             reg = os.path.relpath(dirpath, run_d)
-            if reg_pattern and not reg_pattern.match(reg):
-                continue
+            if reg_pattern:
+                if compile_reg:
+                    try:
+                        reg_pattern = re.compile(reg_pattern)
```

Comment:
When we have a `reg_pattern`, and `compile_reg` is `True`, it means that this will be called N times, N being the length of directories in `run_dirs` times the number of its subdirectories. It we need to compile it, then I think it would be better to do it outside/before the loop starts just once?

Reply:
Ah, very good spot, thanks! Yes, calling it more than once is needless. Actually, this will be solved already in my resolution to the previous review comment, by moving the ``try/except`` block out into ``bin/cylc-print``, where ti will only be called once :+1:.